internal/consul/kv.go

```diff
@@ -112,3 +132,37 @@ func (k *kv) ListMap(prefix, offset string) (map[string]interface{}, error) {
 
 	return result, nil
 }
+
+// Store a map[string]interface data in KV storage under the prefix key
+func (k *kv) PutMap(prefix string, data map[string]interface{}) error {
+	for key, value := range data {
+		switch c := value.(type) {
+		case string:
+			err := k.PutStr(fmt.Sprintf("%s/%s", prefix, key), c)
+			if err != nil {
+				return err
+			}
+		default:
+			mapInterface := make(map[string]interface{})
+			mapstructure.Decode(value, &mapInterface)
+			err := k.PutMap(fmt.Sprintf("%s/%s", prefix, key), mapInterface)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (k *kv) PutStr(prefix string, value string) error {
+	_, err := k.Put(&consulApi.KVPair{
+		Key:   prefix,
+		Value: []byte(value)}, nil)
+
+	if err != nil {
+		return errors.Wrap(err, "Error storing a new value in the KV storage")
+	}
+
+	//log.Printf("Value %s properly stored at %s", value, prefix)
```

Comment:
please make sure not to forget this commented line :)

Reply:
As we don't have a good logging system right now, with different verbosity levels, I just commented this, because it prints hundreds of lines every discovery. It would be a `debug` log in a final version.